pyteal/ast/subroutine.py

```diff
@@ -105,13 +105,6 @@ def _validate_parameter_type(
             # * `invoke` type checks provided arguments against parameter types to catch mismatches.
             return Expr
         else:
-            if not isclass(ptype) and not SubroutineDefinition._is_abi_annotation(
-                ptype
-            ):
-                raise TealInputError(
-                    f"Function has parameter {parameter_name} of declared type {ptype} which is not a class"
-                )
-
             if ptype in (Expr, ScratchVar):
```

Comment:
FYI - [my merges ](https://github.com/algorand/pyteal/pull/310/files) are about to repeat this work.

And optionally: Since you are returning out of the if's you can remove the else's and write the last 2 conditions as:

```python
            if SubroutineDefinition._is_abi_annotation(ptype):
                return abi.type_spec_from_annotation(ptype)
            raise TealInputError(
                f"Function has parameter {parameter_name} of disallowed type {ptype}. "
                f"Only the types {(Expr, ScratchVar, 'ABI')} are allowed"
            )
```